### DISCUSSION TITLE
FP AVD-AWS-0035

### DISCUSSION THREAD

**Discussion by tzurielweisberg** [aquasecurity/trivy#5146] [False Detection]
### IDs

AVD-AWS-0035

### Description

Customer is having an AVD-AWS-0035 in the Supply Chain, but as they insist, they don’t use EFS volume

![image](https://github.com/aquasecurity/trivy/assets/23654690/a013c46e-177e-4e1e-b6e8-de9fd89f77c6)

### Reproduction Steps

```bash
1.
2.
3.
...
```

### Target

Filesystem

### Scanner

Misconfiguration

### Debug Output

```bash
.
```

### Version

```bash
v0.42.0
```

### Checklist

- [ ] Read [the documentation regarding wrong detection](https://aquasecurity.github.io/trivy/dev/community/contribute/discussion/#false-detection)
- [ ] Ran Trivy with `-f json` that shows data sources and confirmed that the security advisory in data sources was correct

**Comment by simar7** [comment]
It would be helpful to see a small snippet with which we can reproduce this issue. 

FYI this rule isn't new as it has existed for several years.

**Comment by eswets** [comment]
Just hit this issue as well, i'll attempt to provide some context. 

On ECS [several different types of data volumes](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/using_data_volumes.html) are supported. In our case, we use a [bind mount](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/specify-bind-mount-config.html) in our task definition.

With this configuration, ECS uses ephemeral storage, which is [by default encrypted](https://docs.aws.amazon.com/AmazonECS/latest/bestpracticesguide/security-fargate.html#security-fargate-ephemeral-storage-encryption) when running on Fargate 1.4.0.

So, I guess at the very least the detection rule should more specifically target EFS volumes (e.g., `efsVolumeConfiguration` object should be present in a volume configuration block). Or perhaps the rule can be expanded (or another rule created) to also evaluate the configurations of the different volume types.
